sdk/core/core-http/src/serializer.ts

```diff
@@ -1231,37 +1348,92 @@ export interface BaseMapper {
   constraints?: MapperConstraints;
 }
 
+/**
+ * Mappers are definitions of the Data models used in the library.
+ * These data models are part of the Operation or Client definitions in the responses or parameters.
+ */
 export type Mapper = BaseMapper | CompositeMapper | SequenceMapper | DictionaryMapper | EnumMapper;
 
+/**
+ * Helps identify how to transform from one set of properties in the source object, to a set of properties in the target object.
```

Comment:
This actually is used to disambiguate discriminated type unions, e.g. when a response shape can be one of many, but there is a 'kind' field or some such that tells them apart and we wish to deserialize the correct subtype.